app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdScreenController.kt

```diff
@@ -26,54 +17,8 @@ class ScanSimpleIdScreenController @Inject constructor(
 
   override fun apply(events: Observable<UiEvent>): ObservableSource<UiChange> {
     val replayedEvents = ReplayUntilScreenIsDestroyed(events)
-        .compose(mergeWithScannedBpPassportCodes())
         .replay()
 
-    return handleScannedBpPassportCodes(replayedEvents)
-  }
-
-  private fun mergeWithScannedBpPassportCodes() = ObservableTransformer<UiEvent, UiEvent> { events ->
-    val scannedBpPassportCodes = events
-        .ofType<ScanSimpleIdScreenQrCodeScanned>()
-        .map { it.text }
-        .map { scannedQrCode ->
-          try {
-            val bpPassportCode = UUID.fromString(scannedQrCode)
-            ValidPassportCode(bpPassportCode)
-          } catch (e: IllegalArgumentException) {
-            InvalidPassportCode
```

Comment:
I see this is not being handled in the update function. IIRC, we just ignore invalid QR codes. Just confirming that ignoring it is the expected behaviour.

Reply:
Yes, ignoring invalid QR codes is the expected behaviour